data/js/tomahawk.js

```diff
@@ -508,14 +509,26 @@ Tomahawk.ajax = function(url, settings) {
 
             return str.join("&");
         };
+        if (typeof settings.data === 'object') {
+            if (settings.encoding == 'form') {
+                settings.data = formEncode(settings.data);
+                settings.content_type = settings.content_type || 'application/x-www-form-urlencoded';
+            } else if (settings.encoding == 'json') {
+                settings.data = JSON.stringify(settings.data);
+                settings.content_type = settings.content_type || 'application/json';
+            } else {
+                throw new Error("Tomahawk.ajax: unknown data encoding requested: " + settings.encoding);
+            }
+        }
 
         if (settings.type.toLowerCase() === 'get') {
-            settings.url += '?' + formEncode(settings.data);
+            settings.url += '?' + settings.data;
```

Comment:
Behaviour is not broken. settings.data gets overwritten with encoded form
in the above block

On Sat, May 2, 2015 at 10:41 AM, Dominik Schmidt <notifications@github.com>
wrote:

> In data/js/tomahawk.js
> <https://github.com/tomahawk-player/tomahawk/pull/296#discussion_r29549172>
> :
>
> >
> >          if (settings.type.toLowerCase() === 'get') {
> > -            settings.url += '?' + formEncode(settings.data);
> > +            settings.url += '?' + settings.data;
>
> Sorry, i missed that you are actually breaking behaviour here. For get
> requests data should be encoded in the url
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/tomahawk-player/tomahawk/pull/296/files#r29549172>.
>